app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java

```diff
@@ -0,0 +1,206 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.UserNotFoundException;
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserFixtures;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserData;
+import com.github.dozermapper.core.Mapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+
+@DisplayName("UserService 클래스의")
+@ExtendWith(MockitoExtension.class)
+@ActiveProfiles("test")
+public class UserServiceTest {
+
+    private final User userAlice = UserFixtures.alice();
+    private final User userBob = UserFixtures.bob();
+    private final UserData validUserDataAlice = UserFixtures.validAliceData();
+    private final UserData validUserDataBob = UserFixtures.validBobData();
+
+    @InjectMocks
+    UserService userService;
+    @Mock
+    UserRepository userRepository;
+    @Mock
+    Mapper mapper;
+
+    @Nested
+    @DisplayName("get 메서드는")
+    class Describe_get {
+
+        @Nested
+        @DisplayName("만약 주어진 ID의 사용자가 저장되어 있을 경우")
+        class Context_with_existed_user_id {
+
+            @BeforeEach
+            void mocking() {
+                given(userRepository.findById(userBob.getId()))
+                        .willReturn(Optional.ofNullable(userBob));
+            }
```

Comment:
테스트를 잘 작성하셨습니다. 이제 한 단계 더 올라가서 생각해 보도록 합시다. `@DisplayName`의 설명과 `mocking` 메소드의 내용을 비교해 보세요. 좋은 테스트가 있는데도 미묘한 괴리감이 느껴집니다.

충분히 많이 생각해 보신 후 아래의 코드를 읽어보세요.

```suggestion
        @Nested
        @DisplayName("만약 주어진 ID의 사용자를 찾을 수 있다면")
        class Context_with_existed_user_id {

            @BeforeEach
            void mocking() {
                given(userRepository.findById(userBob.getId()))
                        .willReturn(Optional.ofNullable(userBob));
            }
```

이제 설명에는 있지만 준비 코드에는 없었던 구체적인 개념 `저장`이 제거되었고, `저장`은 `찾을 수 있는`이라는 고수준 레이어의 표현으로 추상화되었습니다. 따라서 `저장`과 `findById.willReturn`이라는 코드와의 괴리도 감소했습니다.

 `Context`의 설명과 준비 코드가 조금 더 일치하게 된 모습에서 어떤 느낌을 받으셨나요?

Reply:
더 정확하고 이해하기 쉬운 설명이 되는 것 같습니다. 심지어 변경 전 저의 설명은 마치 준비 코드에서 `save()` 메서드를 호출해야 하는 것처럼 비춰질 수도 있단 생각도 듭니다.